java-frontend/src/main/java/org/sonar/plugins/java/api/tree/MethodInvocationTree.java

```diff
@@ -49,6 +49,6 @@ public interface MethodInvocationTree extends ExpressionTree {
 
   Arguments arguments();
 
-  Symbol symbol();
+  Symbol.MethodSymbol symbol();
```

Comment:
This change generates compilation warnings about useless casts to MethodSymbol.
Like in `java-checks/src/main/java/org/sonar/java/checks/aws/AwsConsumerBuilderUsageCheck.java`:
```java
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
        MethodSymbol methodSymbol = (MethodSymbol) mit.symbol();
```
We should remove those casts.
And to be able to remove the maximum of useless casts, we could also change the signature of NewClassTree:
```java
Symbol.MethodSymbol constructorSymbol();
```